src/main/java/com/planning_system/services/utility/TaskUtil.java

```diff
@@ -1,39 +1,33 @@
 package com.planning_system.services.utility;
 
-import com.planning_system.entity.Task;
-import com.planning_system.entity.TaskPriority;
-import com.planning_system.entity.TaskStatus;
-import com.planning_system.handlers.commands.OptionType;
+import com.planning_system.entity.task.Task;
 
 import java.util.Comparator;
 import java.util.Map;
 import java.util.Objects;
 
-import static com.planning_system.handlers.commands.OptionType.ID;
-import static com.planning_system.handlers.commands.OptionType.REVERSE_SORT;
-import static com.planning_system.handlers.commands.OptionType.SORTED;
-import static com.planning_system.services.messages.ServiceErrorMessages.ID_NOT_NUMBER;
-import static com.planning_system.services.messages.ServiceErrorMessages.INVALID_PRIORITY;
+import static com.planning_system.controller.task.TaskParams.ID;
+import static com.planning_system.controller.task.TaskParams.REVERSE_SORT;
+import static com.planning_system.controller.task.TaskParams.SORTED_BY;
 import static com.planning_system.services.messages.ServiceErrorMessages.INVALID_SORTING_TYPE;
-import static com.planning_system.services.messages.ServiceErrorMessages.INVALID_STATUS;
 
 public class TaskUtil {
 
-    public static Comparator<Task> getTaskComparatorFromOptions(Map<OptionType, String> options) {
+    public static Comparator<Task> getTaskComparatorFromParams(Map<String, String> params) {
```

Comment:
It's highly conditional logic here, can you find a way to simplify it?

Reply:
I've changed the logic a bit to get rid of code duplication. Another way to get sorted Tasks could be using Spring JPA custom methods, for instance getAllTasksOrderByData.